src/preact/base-element.js

```diff
@@ -60,7 +60,7 @@ export class PreactBaseElement extends AMP.BaseElement {
     /** @private {boolean} */
     this.scheduledRender_ = false;
 
-    /** @private {!Object} */
+    /** @protected {!Object} */
```

Comment:
IMHO, this should stay `private`.

Reply:
Is it possible to access `context` in `AmpSocialShare` while allowing it to be private here? Otherwise this can go back to `private` when we're able to [do away with context.](https://github.com/ampproject/amphtml/pull/26543#discussion_r392730351)

Reply:
This context exists for one thing only: cross-component rendering management. For instance, this is how any component would communicate to our special `Slot` component. It shouldn't be used for anything else. And when used for this reason, it should use it via `useContext()` and `AmpContext.Provider`. This just doesn't seem to be the right data source or right communication style for your needs.

Reply:
Agreed.